### ISSUE TITLE
Eslint does not recognize "async", but I have specified "ecmaVersion":"8"

### ISSUE THREAD

**Issue by lborgman** [eslint/eslint#11976]
I am not sure if this is a bug or not so I am asking here.

I have specified this in packages.json:

`  "eslintConfig": {
    "parserOptions": {
      "ecmaVersion": "8"
    }
  },
`

I am not sure this is the correct syntax since the docs are a bit unclear to me. 

But I get parsing errors on `async`. What is wrong here?

Ah, and this is during `firebase deploy`. Maybe there is something special there?

**Comment by platinumazure** [comment]
Hi @lborgman, just to be sure, could you try changing "ecmaVersion" to a number? (Not sure if we have an auto-convert somewhere.)

Might also be worth checking the firebase docs to see if there's a way to get debug output, in case it might show how it invokes ESLint.

**Comment by lborgman** [comment]
@platinumazure Thanks. I have checked both `8` and `"8"`. Same result.

The command output contains this: 
`Running command: npm --prefix %RESOURCE_DIR% run lint`

This seems to result in `eslint .` in the `functions` directory (which contains the `node_modules` directory).

In the log file I see this:

10 silly lifecycle functions@~lint: Args: [ '/d /s /c', 'eslint .' ]
11 silly lifecycle functions@~lint: Returned: code: 1  signal: null
12 info lifecycle functions@~lint: Failed to exec lint script
13 verbose stack Error: functions@ lint: `eslint .`
13 verbose stack Exit status 1

**Comment by platinumazure** [comment]
Are you able to execute ESLint from the command line? If so, do you get this same issue? (Make sure to use a locally installed ESLint from node_modules, assuming your firebase command also looks for a local install.)

**Comment by lborgman** [comment]
@platinumazure 
Yes, if I run this I get the same error: `node_modules\.bin\eslint .`

However I have not done `eslin --init`, but I guess that is not import here?

**Comment by lborgman** [comment]
@platinumazure The --print-config option does not work.

`node_modules\.bin\eslint . --print-config`
`Value for 'print-config' of type 'path::String' required.`

**Comment by lborgman** [comment]
@platinumazure The --debug option gives a long output. Those two lines seems relevant:

`eslint:config Using .eslintrc and package.json files +1ms`
`eslint:config Using config from cache +7ms`

**Comment by platinumazure** [comment]
@lborgman You don't need to run `eslint --init` if you already have a config file (which you should, since you are specifying ecmaVersion).

Re: `--print-config`: Apologies, I should have mentioned that you need to supply a path to one of the files to be linted, so that we can calculate any glob-based overrides if they are present in your config.

**Comment by lborgman** [comment]
@platinumazure Here is the --print-config output from the file with `async`:

{
  "globals": {},
  "env": {},
  "rules": {
    "accessor-pairs": "off",
    "array-bracket-newline": "off",
    "array-bracket-spacing": "off",
    "array-callback-return": 1,
    "array-element-newline": "off",
    "arrow-body-style": "off",
    "arrow-parens": "off",
    "arrow-spacing": "off",
    "block-scoped-var": "off",
    "block-spacing": "off",
    "brace-style": "off",
    "callback-return": 2,
    "camelcase": "off",
    "capitalized-comments": "off",
    "class-methods-use-this": "off",
    "comma-dangle": "off",
    "comma-spacing": "off",
    "comma-style": "off",
    "complexity": "off",
    "computed-property-spacing": "off",
    "consistent-return": 1,
    "consistent-this": "off",
    "constructor-super": "error",
    "curly": "off",
    "default-case": "off",
    "dot-location": "off",
    "dot-notation": "off",
    "eol-last": "off",
    "eqeqeq": 2,
    "for-direction": 2,
    "func-call-spacing": "off",
    "func-name-matching": "off",
    "func-names": "off",
    "func-style": "off",
    "function-paren-newline": "off",
    "generator-star-spacing": "off",
    "getter-return": 2,
    "global-require": "off",
    "guard-for-in": "off",
    "handle-callback-err": 2,
    "id-blacklist": "off",
    "id-length": "off",
    "id-match": "off",
    "implicit-arrow-linebreak": "off",
    "indent": "off",
    "indent-legacy": "off",
    "init-declarations": "off",
    "jsx-quotes": "off",
    "key-spacing": "off",
    "keyword-spacing": "off",
    "line-comment-position": "off",
    "linebreak-style": "off",
    "lines-around-comment": "off",
    "lines-around-directive": "off",
    "lines-between-class-members": "off",
    "max-depth": "off",
    "max-len": "off",
    "max-lines": "off",
    "max-nested-callbacks": "off",
    "max-params": "off",
    "max-statements": "off",
    "max-statements-per-line": "off",
    "multiline-comment-style": "off",
    "multiline-ternary": "off",
    "new-cap": "off",
    "new-parens": "off",
    "newline-after-var": "off",
    "newline-before-return": "off",
    "newline-per-chained-call": "off",
    "no-alert": 2,
    "no-array-constructor": "off",
    "no-await-in-loop": 2,
    "no-bitwise": "off",
    "no-buffer-constructor": "off",
    "no-caller": 2,
    "no-case-declarations": "error",
    "no-catch-shadow": 1,
    "no-class-assign": "error",
    "no-compare-neg-zero": 2,
    "no-cond-assign": "error",
    "no-confusing-arrow": "off",
    "no-console": "off",
    "no-const-assign": "error",
    "no-constant-condition": "error",
    "no-continue": "off",
    "no-control-regex": "error",
    "no-debugger": "off",
    "no-delete-var": "error",
    "no-div-regex": "off",
    "no-dupe-args": "error",
    "no-dupe-class-members": "error",
    "no-dupe-keys": "error",
    "no-duplicate-case": "error",
    "no-duplicate-imports": "off",
    "no-else-return": "off",
    "no-empty": "error",
    "no-empty-character-class": "error",
    "no-empty-function": "off",
    "no-empty-pattern": "error",
    "no-eq-null": 2,
    "no-eval": 2,
    "no-ex-assign": "error",
    "no-extend-native": 1,
    "no-extra-bind": 1,
    "no-extra-boolean-cast": "error",
    "no-extra-label": 1,
    "no-extra-parens": "off",
    "no-extra-semi": "error",
    "no-fallthrough": "error",
    "no-floating-decimal": 2,
    "no-func-assign": "error",
    "no-global-assign": "error",
    "no-implicit-coercion": 1,
    "no-implicit-globals": "off",
    "no-implied-eval": "off",
    "no-inline-comments": "off",
    "no-inner-declarations": "error",
    "no-invalid-regexp": "error",
    "no-invalid-this": "off",
    "no-irregular-whitespace": "error",
    "no-iterator": "off",
    "no-label-var": "off",
    "no-labels": "off",
    "no-lone-blocks": "off",
    "no-lonely-if": "off",
    "no-loop-func": 1,
    "no-magic-numbers": "off",
    "no-mixed-operators": "off",
    "no-mixed-requires": "off",
    "no-mixed-spaces-and-tabs": "off",
    "no-multi-assign": "off",
    "no-multi-spaces": "off",
    "no-multi-str": "off",
    "no-multiple-empty-lines": "off",
    "no-native-reassign": "off",
    "no-negated-condition": "off",
    "no-negated-in-lhs": "off",
    "no-nested-ternary": "off",
    "no-new": "off",
    "no-new-func": 2,
    "no-new-object": "off",
    "no-new-require": "off",
    "no-new-symbol": "error",
    "no-new-wrappers": 1,
    "no-obj-calls": "error",
    "no-octal": "error",
    "no-octal-escape": "off",
    "no-param-reassign": "off",
    "no-path-concat": 1,
    "no-plusplus": "off",
    "no-process-env": "off",
    "no-process-exit": "off",
    "no-proto": "off",
    "no-prototype-builtins": "off",
    "no-redeclare": "error",
    "no-regex-spaces": "off",
    "no-restricted-globals": "off",
    "no-restricted-imports": "off",
    "no-restricted-modules": "off",
    "no-restricted-properties": "off",
    "no-restricted-syntax": "off",
    "no-return-assign": "off",
    "no-return-await": "off",
    "no-script-url": "off",
    "no-self-assign": "error",
    "no-self-compare": "off",
    "no-sequences": "off",
    "no-shadow": "off",
    "no-shadow-restricted-names": 2,
    "no-spaced-func": "off",
    "no-sparse-arrays": "error",
    "no-sync": "off",
    "no-tabs": "off",
    "no-template-curly-in-string": 1,
    "no-ternary": "off",
    "no-this-before-super": "error",
    "no-throw-literal": 2,
    "no-trailing-spaces": "off",
    "no-undef": "off",
    "no-undef-init": "off",
    "no-undefined": "off",
    "no-underscore-dangle": "off",
    "no-unexpected-multiline": "error",
    "no-unmodified-loop-condition": "off",
    "no-unneeded-ternary": "off",
    "no-unreachable": "error",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": "error",
    "no-unused-expressions": "off",
    "no-unused-labels": "error",
    "no-unused-vars": "off",
    "no-use-before-define": "off",
    "no-useless-call": "off",
    "no-useless-computed-key": "off",
    "no-useless-concat": "off",
    "no-useless-constructor": "off",
    "no-useless-escape": "error",
    "no-useless-rename": "off",
    "no-useless-return": "off",
    "no-var": "off",
    "no-void": "off",
    "no-warning-comments": "off",
    "no-whitespace-before-property": "off",
    "no-with": "off",
    "nonblock-statement-body-position": "off",
    "object-curly-newline": "off",
    "object-curly-spacing": "off",
    "object-property-newline": "off",
    "object-shorthand": "off",
    "one-var": "off",
    "one-var-declaration-per-line": "off",
    "operator-assignment": "off",
    "operator-linebreak": "off",
    "padded-blocks": "off",
    "padding-line-between-statements": "off",
    "prefer-arrow-callback": 1,
    "prefer-const": "off",
    "prefer-destructuring": "off",
    "prefer-numeric-literals": "off",
    "prefer-promise-reject-errors": 2,
    "prefer-reflect": "off",
    "prefer-rest-params": "off",
    "prefer-spread": "off",
    "prefer-template": "off",
    "quote-props": "off",
    "quotes": "off",
    "radix": "off",
    "require-await": "off",
    "require-jsdoc": "off",
    "require-yield": "error",
    "rest-spread-spacing": "off",
    "semi": "off",
    "semi-spacing": "off",
    "semi-style": "off",
    "sort-imports": "off",
    "sort-keys": "off",
    "sort-vars": "off",
    "space-before-blocks": "off",
    "space-before-function-paren": "off",
    "space-in-parens": "off",
    "space-infix-ops": "off",
    "space-unary-ops": "off",
    "spaced-comment": "off",
    "strict": "off",
    "switch-colon-spacing": "off",
    "symbol-description": "off",
    "template-curly-spacing": "off",
    "template-tag-spacing": "off",
    "unicode-bom": "off",
    "use-isnan": "error",
    "valid-jsdoc": "off",
    "valid-typeof": "error",
    "vars-on-top": "off",
    "wrap-iife": "off",
    "wrap-regex": "off",
    "yield-star-spacing": "off",
    "yoda": "off",
    "promise/always-return": 2,
    "promise/catch-or-return": 2,
    "promise/no-nesting": 1
  },
  "parserOptions": {
    "ecmaVersion": 6
  },
  "plugins": [
    "promise"
  ],
  "extends": "eslint:recommended"
}

**Comment by platinumazure** [comment]
@lborgman The `--print-config` output is offering a valuable clue here. Note that it says `"parserOptions": { "ecmaVersion": 6 }`, which is not what you want.

At this point, one of two things can be happening here in general. One, your config with `"ecmaVersion": 8` is not being pulled in for some reason; or two, your config is being pulled in, but some other config is overriding your ecmaVersion setting.

**Comment by lborgman** [comment]
@platinumazure Yes, sorry, I did not know there was a `.eslintrc.json`. (I did not know where to look, but now I see it.) And that file says `6`.

I commented out that line (I just put `//` at the front of it, which seems to work). Now I get 14 errors telling me that `const` is reserved.

It looks like it does not pick up the version of Node.js from `packages.json`. Could you please tell me how the record in that file should look?

**Comment by g-plane** [comment]
There is no relationship between Node.js version and the `ecmaVersion` field.

**Comment by lborgman** [comment]
@platinumazure Thanks. Sorry, that was a typo. I thought I wrote "does not pickup the version of `eslintConfig` from `packages.json`.

**Comment by mysticatea** [comment]
> Prior to 2.0.0, if a directory contained both an .eslintrc file and a package.json file with ESLint configuration information, the settings from the two files would be merged together. In 2.0.0, only the settings from the .eslintrc.* file are used and the ones in package.json are ignored when both are present. Otherwise, package.json can still be used with ESLint configuration, but only if no other .eslintrc.* files are present.
> https://eslint.org/blog/2016/02/eslint-v2.0.0-released#configuration-cascading-changes

Since ESLint 2, if `.eslintrc` file existed, ESLint ignores `eslintConfig` of package.json.

**Comment by lborgman** [comment]
@mysticatea Thanks. That explains what I see. However I think it would be good if ESLint told this (when it is possible). But maybe that is not possible currently (without breaking things)?

**Comment by platinumazure** [comment]
@lborgman I think we could *maybe* accept a PR which would emit a warning of some kind when we've seen an `.eslintrc.*` file, but also see `eslintConfig` in `package.json`, to at least state that the `eslintConfig` is being ignored due to the presence of config files. Would you like to make a separate issue for that change?

**Comment by platinumazure** [comment]
I'm going to close this issue since the original question has been answered. Please feel free to open an issue or pull request about having ESLint emit a warning if it sees one or more config files, plus a package.json with ESLint configuration. Thanks!

**Comment by ZengTianShengZ** [comment]
i have some problem ,here is my config 

```javascript
module.exports = {
  root: true,
  env: {
    "node": true,
  },
  extends: [
    'eslint:recommended'
  ],
  parser: "babel-eslint",
  parserOptions: {    
    ecmaVersion: 8,
    sourceType: 'module',
    ecmaFeatures: {
      modules: true,
      experimentalObjectRestSpread: true
  }
  },
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'space-before-function-paren': [2, 'never'],
    'camelcase': 'off',
    "semi": [1, 'always'],
      // http://eslint.org/docs/rules
      "quotes": [
          2, 'single', 'avoid-escape'    // http://eslint.org/docs/rules/quotes
      ],
      // http://eslint.org/docs/rules/no-unused-vars
      // 'no-unused-vars': [
      //     1, { vars: 'local', args: 'after-used' }
      // ],
      // "no-unused-vars": "off",
      // "typescript/no-unused-vars": ["error", { "vars": "all", "args": "after-used"}],
      'eol-last': 1,
      'no-debugger': 1,
      'padded-blocks': 1,
      'brace-style': [2, 'stroustrup'],
      'comma-dangle': [2, 'only-multiline'],
      'space-before-function-paren': [2, 'never'],
      'no-multiple-empty-lines': [2, { max: 3 }],
      'keyword-spacing': [
          2, {
              before: true,
              after: true,
              overrides: {
                  'catch': {
                      after: false
                  },
                  'for': {
                      after: false
                  },
                  'if': {
                      after: false
                  },
                  'while': {
                      after: false
                  }
              }
          }
      ],
      indent: [2, 2, {
          MemberExpression: 0
      }],
      eqeqeq: 2,
      curly: 1
  }
}

```


 --print-config output from the file with async:

```json
{
  "globals": {
    "arguments": false,
    "GLOBAL": false,
    "root": false,
    "__dirname": false,
    "__filename": false,
    "Buffer": false,
    "clearImmediate": false,
    "clearInterval": false,
    "clearTimeout": false,
    "console": false,
    "exports": true,
    "global": false,
    "Intl": false,
    "module": false,
    "process": false,
    "queueMicrotask": false,
    "require": false,
    "setImmediate": false,
    "setInterval": false,
    "setTimeout": false,
    "TextDecoder": false,
    "TextEncoder": false,
    "URL": false,
    "URLSearchParams": false,
    "Array": false,
    "ArrayBuffer": false,
    "Boolean": false,
    "constructor": false,
    "DataView": false,
    "Date": false,
    "decodeURI": false,
    "decodeURIComponent": false,
    "encodeURI": false,
    "encodeURIComponent": false,
    "Error": false,
    "escape": false,
    "eval": false,
    "EvalError": false,
    "Float32Array": false,
    "Float64Array": false,
    "Function": false,
    "hasOwnProperty": false,
    "Infinity": false,
    "Int16Array": false,
    "Int32Array": false,
    "Int8Array": false,
    "isFinite": false,
    "isNaN": false,
    "isPrototypeOf": false,
    "JSON": false,
    "Map": false,
    "Math": false,
    "NaN": false,
    "Number": false,
    "Object": false,
    "parseFloat": false,
    "parseInt": false,
    "Promise": false,
    "propertyIsEnumerable": false,
    "Proxy": false,
    "RangeError": false,
    "ReferenceError": false,
    "Reflect": false,
    "RegExp": false,
    "Set": false,
    "String": false,
    "Symbol": false,
    "SyntaxError": false,
    "toLocaleString": false,
    "toString": false,
    "TypeError": false,
    "Uint16Array": false,
    "Uint32Array": false,
    "Uint8Array": false,
    "Uint8ClampedArray": false,
    "undefined": false,
    "unescape": false,
    "URIError": false,
    "valueOf": false,
    "WeakMap": false,
    "WeakSet": false
  },
  "env": {
    "node": true,
    "es6": true
  },
  "rules": {
    "accessor-pairs": "off",
    "array-bracket-newline": "off",
    "array-bracket-spacing": "off",
    "array-callback-return": "off",
    "array-element-newline": "off",
    "arrow-body-style": "off",
    "arrow-parens": "off",
    "arrow-spacing": "off",
    "block-scoped-var": "off",
    "block-spacing": "off",
    "brace-style": [
      2,
      "stroustrup"
    ],
    "callback-return": "off",
    "camelcase": "off",
    "capitalized-comments": "off",
    "class-methods-use-this": "off",
    "comma-dangle": [
      2,
      "only-multiline"
    ],
    "comma-spacing": "off",
    "comma-style": "off",
    "complexity": "off",
    "computed-property-spacing": "off",
    "consistent-return": "off",
    "consistent-this": "off",
    "constructor-super": "error",
    "curly": 1,
    "default-case": "off",
    "dot-location": "off",
    "dot-notation": "off",
    "eol-last": 1,
    "eqeqeq": 2,
    "for-direction": "off",
    "func-call-spacing": "off",
    "func-name-matching": "off",
    "func-names": "off",
    "func-style": "off",
    "function-paren-newline": "off",
    "generator-star-spacing": "off",
    "getter-return": "off",
    "global-require": "off",
    "guard-for-in": "off",
    "handle-callback-err": "off",
    "id-blacklist": "off",
    "id-length": "off",
    "id-match": "off",
    "implicit-arrow-linebreak": "off",
    "indent": [
      2,
      2,
      {
        "MemberExpression": 0
      }
    ],
    "indent-legacy": "off",
    "init-declarations": "off",
    "jsx-quotes": "off",
    "key-spacing": "off",
    "keyword-spacing": [
      2,
      {
        "before": true,
        "after": true,
        "overrides": {
          "catch": {
            "after": false
          },
          "for": {
            "after": false
          },
          "if": {
            "after": false
          },
          "while": {
            "after": false
          }
        }
      }
    ],
    "line-comment-position": "off",
    "linebreak-style": "off",
    "lines-around-comment": "off",
    "lines-around-directive": "off",
    "lines-between-class-members": "off",
    "max-depth": "off",
    "max-len": "off",
    "max-lines": "off",
    "max-nested-callbacks": "off",
    "max-params": "off",
    "max-statements": "off",
    "max-statements-per-line": "off",
    "multiline-comment-style": "off",
    "multiline-ternary": "off",
    "new-cap": "off",
    "new-parens": "off",
    "newline-after-var": "off",
    "newline-before-return": "off",
    "newline-per-chained-call": "off",
    "no-alert": "off",
    "no-array-constructor": "off",
    "no-await-in-loop": "off",
    "no-bitwise": "off",
    "no-buffer-constructor": "off",
    "no-caller": "off",
    "no-case-declarations": "error",
    "no-catch-shadow": "off",
    "no-class-assign": "error",
    "no-compare-neg-zero": "error",
    "no-cond-assign": "error",
    "no-confusing-arrow": "off",
    "no-console": "off",
    "no-const-assign": "error",
    "no-constant-condition": "error",
    "no-continue": "off",
    "no-control-regex": "error",
    "no-debugger": 1,
    "no-delete-var": "error",
    "no-div-regex": "off",
    "no-dupe-args": "error",
    "no-dupe-class-members": "error",
    "no-dupe-keys": "error",
    "no-duplicate-case": "error",
    "no-duplicate-imports": "off",
    "no-else-return": "off",
    "no-empty": "error",
    "no-empty-character-class": "error",
    "no-empty-function": "off",
    "no-empty-pattern": "error",
    "no-eq-null": "off",
    "no-eval": "off",
    "no-ex-assign": "error",
    "no-extend-native": "off",
    "no-extra-bind": "off",
    "no-extra-boolean-cast": "error",
    "no-extra-label": "off",
    "no-extra-parens": "off",
    "no-extra-semi": "error",
    "no-fallthrough": "error",
    "no-floating-decimal": "off",
    "no-func-assign": "error",
    "no-global-assign": "error",
    "no-implicit-coercion": "off",
    "no-implicit-globals": "off",
    "no-implied-eval": "off",
    "no-inline-comments": "off",
    "no-inner-declarations": "error",
    "no-invalid-regexp": "error",
    "no-invalid-this": "off",
    "no-irregular-whitespace": "error",
    "no-iterator": "off",
    "no-label-var": "off",
    "no-labels": "off",
    "no-lone-blocks": "off",
    "no-lonely-if": "off",
    "no-loop-func": "off",
    "no-magic-numbers": "off",
    "no-mixed-operators": "off",
    "no-mixed-requires": "off",
    "no-mixed-spaces-and-tabs": "error",
    "no-multi-assign": "off",
    "no-multi-spaces": "off",
    "no-multi-str": "off",
    "no-multiple-empty-lines": [
      2,
      {
        "max": 3
      }
    ],
    "no-native-reassign": "off",
    "no-negated-condition": "off",
    "no-negated-in-lhs": "off",
    "no-nested-ternary": "off",
    "no-new": "off",
    "no-new-func": "off",
    "no-new-object": "off",
    "no-new-require": "off",
    "no-new-symbol": "error",
    "no-new-wrappers": "off",
    "no-obj-calls": "error",
    "no-octal": "error",
    "no-octal-escape": "off",
    "no-param-reassign": "off",
    "no-path-concat": "off",
    "no-plusplus": "off",
    "no-process-env": "off",
    "no-process-exit": "off",
    "no-proto": "off",
    "no-prototype-builtins": "off",
    "no-redeclare": "error",
    "no-regex-spaces": "error",
    "no-restricted-globals": "off",
    "no-restricted-imports": "off",
    "no-restricted-modules": "off",
    "no-restricted-properties": "off",
    "no-restricted-syntax": "off",
    "no-return-assign": "off",
    "no-return-await": "off",
    "no-script-url": "off",
    "no-self-assign": "error",
    "no-self-compare": "off",
    "no-sequences": "off",
    "no-shadow": "off",
    "no-shadow-restricted-names": "off",
    "no-spaced-func": "off",
    "no-sparse-arrays": "error",
    "no-sync": "off",
    "no-tabs": "off",
    "no-template-curly-in-string": "off",
    "no-ternary": "off",
    "no-this-before-super": "error",
    "no-throw-literal": "off",
    "no-trailing-spaces": "off",
    "no-undef": "error",
    "no-undef-init": "off",
    "no-undefined": "off",
    "no-underscore-dangle": "off",
    "no-unexpected-multiline": "error",
    "no-unmodified-loop-condition": "off",
    "no-unneeded-ternary": "off",
    "no-unreachable": "error",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": "error",
    "no-unused-expressions": "off",
    "no-unused-labels": "error",
    "no-unused-vars": "error",
    "no-use-before-define": "off",
    "no-useless-call": "off",
    "no-useless-computed-key": "off",
    "no-useless-concat": "off",
    "no-useless-constructor": "off",
    "no-useless-escape": "error",
    "no-useless-rename": "off",
    "no-useless-return": "off",
    "no-var": "off",
    "no-void": "off",
    "no-warning-comments": "off",
    "no-whitespace-before-property": "off",
    "no-with": "off",
    "nonblock-statement-body-position": "off",
    "object-curly-newline": "off",
    "object-curly-spacing": "off",
    "object-property-newline": "off",
    "object-shorthand": "off",
    "one-var": "off",
    "one-var-declaration-per-line": "off",
    "operator-assignment": "off",
    "operator-linebreak": "off",
    "padded-blocks": 1,
    "padding-line-between-statements": "off",
    "prefer-arrow-callback": "off",
    "prefer-const": "off",
    "prefer-destructuring": "off",
    "prefer-numeric-literals": "off",
    "prefer-promise-reject-errors": "off",
    "prefer-reflect": "off",
    "prefer-rest-params": "off",
    "prefer-spread": "off",
    "prefer-template": "off",
    "quote-props": "off",
    "quotes": [
      2,
      "single",
      "avoid-escape"
    ],
    "radix": "off",
    "require-await": "off",
    "require-jsdoc": "off",
    "require-yield": "error",
    "rest-spread-spacing": "off",
    "semi": [
      1,
      "always"
    ],
    "semi-spacing": "off",
    "semi-style": "off",
    "sort-imports": "off",
    "sort-keys": "off",
    "sort-vars": "off",
    "space-before-blocks": "off",
    "space-before-function-paren": [
      2,
      "never"
    ],
    "space-in-parens": "off",
    "space-infix-ops": "off",
    "space-unary-ops": "off",
    "spaced-comment": "off",
    "strict": "off",
    "switch-colon-spacing": "off",
    "symbol-description": "off",
    "template-curly-spacing": "off",
    "template-tag-spacing": "off",
    "unicode-bom": "off",
    "use-isnan": "error",
    "valid-jsdoc": "off",
    "valid-typeof": "error",
    "vars-on-top": "off",
    "wrap-iife": "off",
    "wrap-regex": "off",
    "yield-star-spacing": "off",
    "yoda": "off"
  },
  "parserOptions": {
    "ecmaFeatures": {
      "globalReturn": true,
      "modules": true,
      "experimentalObjectRestSpread": true
    },
    "ecmaVersion": 8,
    "sourceType": "module"
  },
  "root": true,
  "extends": [
    "eslint:recommended"
  ],
  "parser": "/Users/tiansheng/Documents/tiansheng/guess-stock-server/node_modules/babel-eslint/lib/index.js"
}
```

But I get parsing errors on async.


```
Parsing error: Unexpected token

  2 | 
  3 | export default class HomeController extends Controller {
> 4 |   public async index() {
    |          ^
  5 |     const { ctx } = this;
  6 |     ctx.body = await ctx.service.test.sayHi('egg');
  7 |   }eslint
Peek Problem
```

**Comment by platinumazure** [comment]
Hi @ZengTianShengZ, it looks like you're using babel-eslint so your issue seems different than OP's. It looks like ESLint is correctly using babel-eslint as the parser, so this might be an issue with babel-eslint.

I also see you are using the `public` token. I don't think that's standard JavaScript. If babel-eslint supports that, you may need to set a parser option in order to make sure the `public` token is recognized. If you remove `public` from the method signature, does ESLint work/parse? If so, you need to read the babel-eslint docs to see if there is a parser option that will recognize `public`.

In either case, the issue is related to babel-eslint. Thus, please open an issue on [their repository](https://github.com/babel/babel-eslint) if you get stuck. Thanks!
